### PR TITLE
[TTAHUB-633] Improve display of approval process

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

A few improvements to the approval process

- Enumerate approvers on the my alerts table
- Changes some button labels ("choose report status" becomes "choose approval status", "re-submit report" becomes "update"
- Fixes to the icons & status badges on the table to make them more accurate

## How to test

- Create a report with two approvers
- The report should say "submitted" in the table and you should see the approvers listed
- As the first approver on the report, approve the report. 
- The "you have reviewed" notification appears above the myalerts table, the icon indicates that you approved, and the badge status says "reviewed"
- As the second approver, mark the report as "needs action" 
- The "you have reviewed" notification appears above the myalerts table, the icon indicates that you marked it as needs action, and the badge status says "needs action"
- Update the report and re-submit it for approval
- Approve the report

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-633

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/services/activityReports.js [hunk 1]
```diff
@@ -850,11 +850,16 @@ export async function activityReportAlerts(userId, {
...
-              { calculatedStatus: REPORT_STATUSES.NEEDS_ACTION },
-            ],
-            '$approvers.userId$': userId,
+            calculatedStatus: {
+              [Op.in]: [
+                REPORT_STATUSES.SUBMITTED,
+                REPORT_STATUSES.NEEDS_ACTION,
+              ],
+            },
+            id: {
```
Comment (thewatermethod) [inline src/services/activityReports.js:859]: the changes to this file were necessary as it seems there was name collision happening with the other literals and the joined table aliases

**Review by AdamAdHocTeam — COMMENTED** [review]
After initially submitting the report and returning to the AR landing I don't see any approvers listed.

image

File: frontend/src/pages/ActivityReport/Pages/Review/Approver/Review.js [hunk 2]
```diff
@@ -115,7 +115,7 @@ const Review = ({
         ) : <div className="margin-bottom-3" />}
         <ApproverStatusList approverStatus={approverStatusList} />
         {hasIncompletePages && <IncompletePages incompletePages={incompletePages} />}
-        <Button disabled={hasIncompletePages} type="submit">{hasBeenReviewed ? 'Re-submit' : 'Submit'}</Button>
+        <Button disabled={hasIncompletePages} type="submit">{hasBeenReviewed ? 'Update report' : 'Submit'}</Button>
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/ActivityReport/Pages/Review/Approver/Review.js:118]: Do you think going from Re-submit to 'update' will confuse users into thinking the report wont be submitted.

File: frontend/src/pages/Landing/MyAlerts.js [hunk 3]
```diff
@@ -68,12 +100,40 @@ export function ReportsRow({ reports, removeAlert, message }) {
...
+      displayStatus = 'Needs action';
+      statusClassName = `smart-hub--table-tag-status smart-hub--status-${REPORT_STATUSES.NEEDS_ACTION}`;
     }
+
+    if (
+      calculatedStatus !== REPORT_STATUSES.NEEDS_ACTION
+      && approvers && approvers.length > 0
+      && approvers.some((a) => a.status === APPROVER_STATUSES.APPROVED)
+    ) {
+      displayStatus = 'Reviewed';
```
[the remark this answers is not in the document]
Comment (thewatermethod) [reply frontend/src/pages/Landing/MyAlerts.js:115]: Kind of irrelevant

File: frontend/src/pages/ActivityReport/Pages/Review/Approver/Review.js (hunk 2 above)
Comment (AdamAdHocTeam) [reply frontend/src/pages/ActivityReport/Pages/Review/Approver/Review.js:118]: what could go wrong ....

**Comment by thewatermethod** [comment]
> After initially submitting the report and returning to the AR landing I don't see any approvers listed.

This is how it behaves today (see below for a submitted report with 3 approvers), this is due to the hooks firing after save, fixing this is outside of scope of this PR.

Screenshot 2023-10-23 at 3 30 33 PM

**Comment by AdamAdHocTeam** [comment]
> > After initially submitting the report and returning to the AR landing I don't see any approvers listed.
> 
> @AdamAdHocTeam
> 
> This is how it behaves today (see below for a submitted report with 3 approvers), this is due to the hooks firing after save, fixing this is outside of scope of this PR.
> 
> Screenshot 2023-10-23 at 3 30 33 PM

Lets create a ticket to fix this issue.

**Review by AdamAdHocTeam — APPROVED** [review]
Overall looks good and works.

**Comment by thewatermethod** [comment]
> Lets create a ticket to fix this issue

@AdamAdHocTeam I don't think it's easily fixable, I'm just going to leave it, as we've not received any feedback about this

File: src/services/activityReports.js [hunk 4]
```diff
@@ -866,7 +871,15 @@ export async function activityReportAlerts(userId, {
                 ],
               },
               {
-                [Op.or]: [{ userId }, { '$activityReportCollaborators->user.id$': userId }],
```
Comment (GarrettEHill) [inline src/services/activityReports.js:869]: Was this not working?

File: src/services/activityReports.js [hunk 5]
```diff
@@ -850,11 +850,16 @@ export async function activityReportAlerts(userId, {
         [Op.and]: scopes,
         [Op.or]: [
           {
-            [Op.or]: [
-              { calculatedStatus: REPORT_STATUSES.SUBMITTED },
-              { calculatedStatus: REPORT_STATUSES.NEEDS_ACTION },
-            ],
-            '$approvers.userId$': userId,
```
Comment (GarrettEHill) [inline src/services/activityReports.js:857]: Was this not working?

File: src/services/activityReports.js [hunk 6]
```diff
@@ -895,9 +908,6 @@ export async function activityReportAlerts(userId, {
         sequelize.literal(
           `(SELECT "Recipients".name as recipientName FROM "Recipients" INNER JOIN "ActivityRecipients" ON "ActivityReport"."id" = "ActivityRecipients"."activityReportId" JOIN "Grants" ON "Grants"."id" = "ActivityRecipients"."grantId" AND "Recipients"."id" = "Grants"."recipientId" order by recipientName ${sortDir} limit 1)`,
         ),
-
-        // eslint-disable-next-line quotes
-        [sequelize.literal(`(SELECT  CASE WHEN COUNT(1) = 0 THEN '0' ELSE  CONCAT(SUM(CASE WHEN COALESCE("ActivityReportApprovers".status,'needs_action') = 'approved' THEN 1 ELSE 0 END), ' of ', COUNT(1)) END FROM "ActivityReportApprovers" WHERE "ActivityReportApprovers"."activityReportId" = "ActivityReport"."id" AND "deletedAt" IS NULL limit 1)`), 'pendingApprovals'],
```
Comment (GarrettEHill) [inline src/services/activityReports.js:900]: This would be more efficient with the filter method over the case method

**Pull request merged**